#!/usr/bin/env python

from dynamic_reconfigure.parameter_generator_catkin import *
import math

PACKAGE = "laser_filters"

gen = ParameterGenerator()

##### gen.add("name", paramtype, level, description, default, min, max)

gen.add("use_message_range_limits", bool_t, 0, "use the limits from laser scan msg", False)
gen.add("lower_threshold", double_t, 0, "scans below this threshold is removed", 0.5, 0, 20)
gen.add("upper_threshold", double_t, 0, "scans above this threshold is removed", 5, 0, 20)
gen.add("lower_replacement_value", double_t, 0, "use the limits from laser scan msg", -20, -30, 30)
gen.add("upper_replacement_value", double_t, 0, "use the limits from laser scan msg", 20, -30, 30)
gen.add("switch_", bool_t, 0, "Switches filter on or off", True)

exit(gen.generate(PACKAGE, "laser_filters", "RangeFilter"))